py-polars/polars/interchange/dataframe.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from itertools import accumulate
+from typing import TYPE_CHECKING
+
+from polars.interchange.column import PolarsColumn
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from typing import Any
+
+    from polars import DataFrame
+
+
+class PolarsDataFrame:
+    """A dataframe object backed by a Polars DataFrame."""
+
+    def __init__(self, df: DataFrame, *, allow_copy: bool = True):
```

Comment:
Do we need to check if all chunks in the dataframe align? Is that an invariant?

Reply:
We check for this if the consumer calls `get_chunks`. If the chunks don't align, we either rechunk or raise an error if `allow_copy=False`.

Relevant code:

https://github.com/pola-rs/polars/blob/e2c9ebe6823a278dd34cb0ecaf86f35963f1e1e9/py-polars/polars/interchange/dataframe.py#L163-L168